src/Extensions/Abstractions/ReplayableLogs/TimedScopeDefinition.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+
+namespace Microsoft.Omex.Extensions.Abstractions
+{
+	/// <summary>
+	/// Store Timed Scope name
+	/// </summary>
+	public readonly struct TimedScopeDefinition
+	{
+		/// <summary>
+		/// Timed Scope name
+		/// </summary>
+		public string Name { get; }
+
+
+		/// <summary>
+		/// TimedScopeDefinition constructor
+		/// </summary>
+		/// <param name="name">Timed Scope name</param>
+		/// <exception cref="ArgumentException">Thrown when name null or empty</exception>
+		public TimedScopeDefinition(string name) =>
+			Name = string.IsNullOrWhiteSpace(name)
+				? throw new ArgumentException("TimedScope name must not be empty or null") // Activity won't be stated in case of an empty string
```

Comment:
Technically, the comment is slightly off as we're also considering strings containing only whitespace

Reply:
I think it's fine to keep it like this